### PR TITLE
Add eslintrc for CodeFactor.io

### PR DESCRIPTION
**Pull request by Holzhaus**
This adds a custom `.eslint.json` that's better suited for linting controller scripts. It will automatically be picked up by Codefactor.io and follow [our JS coding conventions](https://www.mixxx.org/wiki/doku.php/contributing_mappings#coding_conventions_for_javascript) more closely than the default config.

You can also use it locally, i.e.:
```bash
# Lint single file
$ eslint res/controllers/Roland_DJ-505-scripts.js
# Lint all *.js files in the repository
$ eslint .
```

### CODE REVIEW

**Comment by Holzhaus** [comment]
By the way, I also auto-fixed a lot of the new issues, but I didn't know if such a mass-change of controller scripts is desired. I pushed it to another branch: https://github.com/mixxxdj/mixxx/commit/82606870a84c6fdb9ae134200382bed46de5f70e

**Comment by Holzhaus** [comment]
I removed all style fixes in `*.js*` files and rebased on 2.2.

@Be-ing I'd like to merge this ASAP. The improved linter settings should make reviewing controller script PRs much easier.

**Comment by uklotzde** [comment]
I like the Rust naming convention of prepending an underscore to unused variables and args:

```diff
-"no-unused-vars": ["error", { "vars": "all", "args": "none" }],
+"no-unused-vars": ["error", { "vars": "all", "args": "all" }, { "argsIgnorePattern": "^_" }],
```

Implicitly ignoring all unused args may occlude obsolete and redundant code.

**Comment by uklotzde** [comment]
...maybe we could then stay with the default settings for vars ("all") and args ("after-used"):

```
"no-unused-vars": ["error", { "argsIgnorePattern": "^_" }],
```

File: .eslintrc.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "sourceType": "script"
   },
   "rules": {
-    "no-unused-vars": ["error", { "vars": "all", "args": "none" }],
+    "no-unused-vars": ["error", { "argsIgnorePattern": "^_" }],
```
Comment (uklotzde) [inline .eslintrc.json]: :+1: With this convention we can declare all available callback function parameters in controller scripts, even if we don't need them.

**Comment by Holzhaus** [comment]
I added some more code style enforcements regarding braces and spacing. Yes, this raises a lot of issues with our existing codebase, but it will make sure that all new controller scripts follow a consistent code style. CodeFactor will only complain about new issues introduced by a PR, not those that are already present, so this shouldn't be a problem.

@uklotzde @Be-ing Please review the rules again. I think all of them make sense -  let me know if you agree with them.

**Comment by uklotzde** [comment]
LGTM. Even VS Code picked up the rules.

**Review by uklotzde — APPROVED** [review]

**Comment by Holzhaus** [comment]
Since this is just a linter config improvement that will make controller mapping reviews much easier and can easily be improved further later on, I'll merge this now.

**Comment by uklotzde** [comment]
:+1: I was unsure if you were waiting for @Be-ing's response and didn't merge it myself only for this reason.

**Pull request merged**